packages/gatsby/src/commands/build.js

```diff
@@ -144,7 +169,17 @@ module.exports = async function build(program: BuildArgs) {
   // we need to save it again to make sure our latest state has been saved
   await db.saveState()
 
-  const pagePaths = [...store.getState().pages.keys()]
+  let pagePaths = [...store.getState().pages.keys()]
+
+  // Rebuild subset of pages if user opt into GATSBY_EXPERIMENTAL_PAGE_BUILD_ON_DATA_CHANGES
+  // if there were no source files (for example components, static queries, etc) changes since last build, otherwise rebuild all pages
+  if (process.env.GATSBY_EXPERIMENTAL_PAGE_BUILD_ON_DATA_CHANGES) {
+    pagePaths =
+      cachedWebpackCompilationHash !== store.getState().webpackCompilationHash
+        ? [...store.getState().pages.keys()]
```

Comment:
this line is the same like in line 172
> `#172: let pagePaths = [...store.getState().pages.keys()]`

can data from line 172 reused?

Reply:
🤦‍♂ Yep spot chnaged this